New 1.4.4 Info Accessory Hooks

MutantWafflez opened the pull request:
### What is the new feature?

Added 3 new hooks: `ModPlayer.ResetInfoAccessories()`, `ModPlayer.RefreshInfoAccessoriesFromTeamPlayers()` and `Mod/GlobalItem.UpdateInfoAccessory()`.

### Why should this be part of tModLoader?

Takes advantage of the new 1.4.4 feature/fix that keeps info accessories updating, even when the game is paused. Furthermore, allows modders to let players of the same teams share info accessory if they are close enough; a feature that vanilla has for vanilla info accessories, but not yet a feature for modded ones.

### Are there alternative designs?

Probably, but this was the design as advised by Chicken Bones.

### Sample usage for the new feature

Check ExampleMod's `ExampleInfoDisplay`, `ExampleInfoDisplayPlayer`, and `ExampleInfoAccessory`.

MutantWafflez committed:
ModPlayer hooks

MutantWafflez committed:
Mod/GlobalItem Hooks

MutantWafflez committed:
ExampleMod & Comment Updates, call order fix

MutantWafflez committed:
Remove task from todo list

MutantWafflez committed:
Resprite of ExampleInfoAccessory (by Rijam#3788)

JavidPack commented:
Would it be useful to mention ItemID.Sets.WorksInVoidBag? I imagine someone might want to disable that for some specific item

Chicken-Bones commented on patches/tModLoader/Terraria/Player.cs.patch:
```diff
@@ -1350,36 +1350,80 @@
++		ItemLoader.UpdateInfoAccessory(item, this);
+ 	}
+ 
+ 	public void RefreshInfoAccsFromItemType(int accType)
```
make `private` please, since it doesn't call the hook. Alternatively, no need to actually split the method. Just declare `int accType`

MutantWafflez committed:
Request Changes Made

- ExampleInfoAccessory now works in Void bag w/ comments
- Made vanilla RefreshInfoAccsFromItemType(int accType) private. shadowed by tML method
